### PR TITLE
Avoid caching WDA after server upgrade

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
This PR depends on https://github.com/appium/WebDriverAgent/pull/111

The idea is to decide caching based on carthage folder timestamp. This folder is only created once on first module install/upgrade, so we can use its timestamp to distinguish an outdated WDA build from the current one.

### CODE REVIEW

**Comment by mykola-mokhnach** [comment]
@KazuCocoa Added tests

**Comment by mykola-mokhnach** [comment]
The script for revision hash creation might look like

```
    "preversion": "git rev-parse HEAD ./WebDriverAgent | head -n 1 > wda.hash",
    "version": "git add -A wda.hash",
```

**Comment by mykola-mokhnach** [comment]
@imurchie can you please take a look at the PR again? I've changed the logic there a bit, so the commit hash is not used anymore

File: lib/wda/utils.js [hunk 1]
```diff
@@ -71,7 +71,7 @@ async function checkForDependencies (bootstrapPath, useSsl = false) {
       'and make sure the root folder, where carthage binary is installed, is present in PATH environment variable. ' +
       `The current PATH value: '${process.env.PATH ? process.env.PATH : "<not defined for the Appium process>"}'`);
   }
-  const carthageRoot = `${bootstrapPath}/Carthage`;
+  const carthageRoot = `${bootstrapPath}/${CARTHAGE_ROOT}`;
```
Comment (imurchie) [inline lib/wda/utils.js]: This should probably be `path.resolve` as well (like line 283 below)

File: lib/wda/webdriveragent.js [hunk 2]
```diff
@@ -275,29 +275,38 @@ class WebDriverAgent {
...
-      log.debug(`WDA is currently not running`);
+      log.debug('WDA is currently not running. There is nothing to cache');
       return;
     }
 
-    if (util.hasValue(status.build.productBundleIdentifier)) {
-      currentAppBundleID = status.build.productBundleIdentifier;
+    const {
+      productBundleIdentifier,
+      upgraded_at,
```
Comment (imurchie) [inline lib/wda/webdriveragent.js]: This should be translated into camel case for consistency.

Comment (mykola-mokhnach) [reply lib/wda/webdriveragent.js]: https://github.com/appium/WebDriverAgent/pull/112

File: lib/wda/utils.js (hunk 1 above)
Comment (mykola-mokhnach) [reply lib/wda/utils.js]: changed

**Pull request merged**